OpenEdXMobile/src/main/java/org/edx/mobile/notifications/services/NotificationService.java

```diff
@@ -42,72 +21,8 @@ public void onNewToken(String s) {
 
-        // TODO: Implement function in PushLinkManager to handle foreground notification
-        //PushLinkManager.INSTANCE.handleForgroundFCMNotification(*)
-
-        if (!environment.getConfig().areFirebasePushNotificationsEnabled()) {
-            // Do not process Notifications when they are disabled.
-            return;
-        }
-
-        if (remoteMessage.getNotification() != null) {
-            logger.debug(
-                    "Message Notification Body: " + remoteMessage.getNotification().getBody()
-            );
-        }
-
-        // Send the message upstream to generate a system notification
-        sendNotification(
-                remoteMessage.getNotification().getTitle(),
-                remoteMessage.getNotification().getBody()
-        );
-    }
-
-    /**
-     * Create and show a simple notification containing the message data.
-     */
-    private void sendNotification(String title, String messageBody) {
-        final Intent intent = new Intent(this, SplashActivity.class);
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-        final PendingIntent pendingIntent = PendingIntent.getActivity(
-                this,
-                0 /* Request code */,
-                intent,
-                PendingIntent.FLAG_ONE_SHOT);
-
-        final Uri defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
-        // Build out the Notification and set the intent to direct the user to the application
-        NotificationCompat.Builder notificationBuilder =
-                new NotificationCompat.Builder(this, DEFAULT_NOTIFICATION_CHANNEL_ID)
-                        .setSmallIcon(R.drawable.ic_notification)
-                        .setContentTitle(title)
-                        .setContentText(messageBody)
-                        .setAutoCancel(true)
-                        .setSound(defaultSoundUri)
-                        .setContentIntent(pendingIntent);
-
-        final NotificationManager notificationManager =
-                (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
-
-        // Since android Oreo notification channel is needed.
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            final NotificationChannel channel = new NotificationChannel(
-                    DEFAULT_NOTIFICATION_CHANNEL_ID,
-                    getResources().getString(R.string.default_notification_channel_title),
-                    NotificationManager.IMPORTANCE_DEFAULT
-            );
-            try {
-                notificationManager.createNotificationChannel(channel);
-            } catch (NullPointerException ex) {
-                logger.error(ex);
-                return;
-            }
-        }
-        try {
-            notificationManager.notify(NOTIFICATION_ID, notificationBuilder.build());
-        } catch (NullPointerException ex) {
-            logger.error(ex);
-            return;
```

Comment:
We are using it, we shall follow the same patter for them as well i-e- show alert dialog in case app is in foreground state.
Update some code related to that.